--- a/src/cds_resource_manager.cc	2018-03-20 02:41:19.000000000 +0100
+++ b/src/cds_resource_manager.cc	2018-05-02 15:34:51.754476482 +0200
@@ -439,6 +439,18 @@
                 }
             }
 
+        if (mimeType.startsWith(_("video"))){
+            extend = extend + _(D_PROFILE) + "=";
+            if (mimeType.find("mp4") > 0)
+                extend = extend + _(D_PN_AVC_MP4_EU) + ";";
+            else if ((mimeType.find("mkv") > 0) || (mimeType.find("x-matroska") > 0))
+                extend = extend + _(D_PN_MKV) + ";";
+            else if (mimeType.find("avi") > 0)
+                extend = extend + _(D_PN_AVI) + ";";
+            else /* if ((mimeType.find("mpeg") > 0) */
+                extend = extend + _(D_PN_MPEG_PS_PAL) + ";";
+        }
+
         // we do not support seeking at all, so 00
         // and the media is converted, so set CI to 1
         if (!isExtThumbnail && transcoded)
--- a/src/common.h	2018-03-20 02:41:19.000000000 +0100
+++ b/src/common.h	2018-05-02 14:47:51.322360803 +0200
@@ -83,6 +83,11 @@
     #define D_JPEG_TN                   "JPEG_TN"
     #define D_JPEG_SM_ICO               "JPEG_SM_ICO"
     #define D_JPEG_LRG_ICO              "JPEG_LRG_ICO"
+    #define D_PN_MPEG_PS_PAL            "MPEG_PS_PAL"
+    #define D_PN_MKV                    "MKV"
+    #define D_PN_AVC_MP4_EU             "AVC_MP4_EU"
+    #define D_PN_AVI                    "AVI"
+
 #endif
 // fixed CdsObjectIDs
 #define CDS_ID_BLACKHOLE                -1
