diff -Nur a/src/file_request_handler.cc b/src/file_request_handler.cc
--- a/src/file_request_handler.cc	2018-11-30 17:28:15.435905000 +0100
+++ b/src/file_request_handler.cc	2018-11-30 18:33:29.319143497 +0100
@@ -53,7 +53,9 @@
 
 void FileRequestHandler::get_info(IN const char* filename, OUT UpnpFileInfo* info)
 {
+#if UPNP_VERSION <= 10803
     HttpProtocolHelper httpProtocolHelper;
+#endif
     log_debug("start\n");
 
     String mimeType;
@@ -291,8 +291,12 @@
 #endif
 
     if (string_ok(header)) {
+#if UPNP_VERSION > 10803
+         UpnpFileInfo_set_ExtraHeaders(info, ixmlCloneDOMString(header.c_str()));
+#else
         std::string finalHeader = httpProtocolHelper.finalizeHttpHeader(header.c_str());
         UpnpFileInfo_set_ExtraHeaders(info, ixmlCloneDOMString(finalHeader.c_str()));
+#endif
     }
 
     UpnpFileInfo_set_LastModified(info, statbuf.st_mtime);
