--- a/pyproject.toml
+++ b/pyproject.toml
@@ -6,7 +6,6 @@ requires = [
     "wheel",
     # Must be kept in sync with the `setup_requirements` in `setup.py`
     "cffi>=1.12; platform_python_implementation != 'PyPy'",
-    "setuptools-rust>=0.11.4",
 ]
 build-backend = "setuptools.build_meta"
 
--- a/setup.py
+++ b/setup.py
@@ -11,7 +11,7 @@ import sys
 from setuptools import find_packages, setup
 
 try:
-    from setuptools_rust import RustExtension
+    pass
 except ImportError:
     print(
         """
@@ -49,20 +49,6 @@
         cffi_modules=[
             "src/_cffi_src/build_openssl.py:ffi",
         ],
-        rust_extensions=[
-            RustExtension(
-                "cryptography.hazmat.bindings._rust",
-                "src/rust/Cargo.toml",
-                py_limited_api=True,
-                # Enable abi3 mode if we're not using PyPy.
-                features=(
-                    []
-                    if platform.python_implementation() == "PyPy"
-                    else ["pyo3/abi3-py36"]
-                ),
-                rust_version=">=1.48.0",
-            )
-        ],
     )
 except:  # noqa: E722
     # Note: This is a bare exception that re-raises so that we don't interfere

