diff -Nur a/core/adbd/adb.c b/core/adbd/adb.c
--- a/core/adbd/adb.c	2020-08-12 04:06:52.000000000 +0200
+++ b/core/adbd/adb.c	2024-11-25 00:35:07.259572955 +0100
@@ -25,6 +25,8 @@
 #include <string.h>
 #include <time.h>
 #include <sys/time.h>
+#include <sys/capability.h>
+#include <sys/prctl.h>
 #include <stdint.h>
 
 #include "sysdeps.h"
@@ -38,7 +38,6 @@
 #if !ADB_HOST
 #include "android_filesystem_config.h"
 #include <linux/capability.h>
-#include <linux/prctl.h>
 #include <sys/mount.h>
 #else
 #include "usb_vendors.h"
diff -Nur a/core/adb/adb_auth_client.c b/core/adb/adb_auth_client.c
--- a/core/adb/adb_auth_client.c	2020-08-12 04:06:52.000000000 +0200
+++ b/core/adb/adb_auth_client.c	2024-11-25 01:37:46.792409915 +0100
@@ -45,6 +45,7 @@
 static fdevent listener_fde;
 static int framework_fd = -1;

+int b64_pton(char const *src, unsigned char *target, size_t targsize);

 static void read_keys(const char *file, struct listnode *list)
 {
diff -Nur a/core/adbd/adb_auth_client.c b/core/adbd/adb_auth_client.c
--- a/core/adbd/adb_auth_client.c	2020-08-12 04:06:52.000000000 +0200
+++ b/core/adbd/adb_auth_client.c	2024-11-25 01:37:29.606311044 +0100
@@ -45,6 +45,7 @@
 static fdevent listener_fde;
 static int framework_fd = -1;

+int b64_pton(char const *src, unsigned char *target, size_t targsize);

 static void read_keys(const char *file, struct listnode *list)
 {
