diff -Nur a/common/autoconf/platform.m4 b/common/autoconf/platform.m4
--- a/common/autoconf/platform.m4	2017-07-05 19:25:40.000000000 +0200
+++ b/common/autoconf/platform.m4	2019-09-03 02:46:22.528995108 +0200
@@ -432,9 +432,9 @@
   # keep track of these additions in ADDED_CFLAGS etc. These
   # will later be checked to make sure only controlled additions
   # have been made to CFLAGS etc.
-  ADDED_CFLAGS=" -m${OPENJDK_TARGET_CPU_BITS}"
-  ADDED_CXXFLAGS=" -m${OPENJDK_TARGET_CPU_BITS}"
-  ADDED_LDFLAGS=" -m${OPENJDK_TARGET_CPU_BITS}"
+  # ADDED_CFLAGS=" -m${OPENJDK_TARGET_CPU_BITS}"
+  # ADDED_CXXFLAGS=" -m${OPENJDK_TARGET_CPU_BITS}"
+  # ADDED_LDFLAGS=" -m${OPENJDK_TARGET_CPU_BITS}"
 
   CFLAGS="${CFLAGS}${ADDED_CFLAGS}"
   CXXFLAGS="${CXXFLAGS}${ADDED_CXXFLAGS}"
