diff -Nur a/hotspot/make/linux/makefiles/adjust-mflags.sh b/hotspot/make/linux/makefiles/adjust-mflags.sh
--- a/hotspot/make/linux/makefiles/adjust-mflags.sh	2017-07-05 19:25:40.000000000 +0200
+++ b/hotspot/make/linux/makefiles/adjust-mflags.sh	2019-09-04 10:23:41.066900411 +0200
@@ -57,6 +57,7 @@
 esac
 
 MFLAGS=$1
+MFLAGSORG=$1
 HOTSPOT_BUILD_JOBS=${2-}
 
 # Normalize any -jN argument to the form " -j${HBJ}"
@@ -84,4 +85,4 @@
     esac;;
 esac
 
-echo "${MFLAGS}"
+echo "${MFLAGSORG}"
